src/main/resources/resource/WebGui/app/service/js/WebkitSpeechRecognitionGui.js

```diff
@@ -1,239 +1,290 @@
 angular.module('mrlapp.service.WebkitSpeechRecognitionGui', []).controller('WebkitSpeechRecognitionGuiCtrl', ['$scope', '$log', 'mrl', function($scope, $log, mrl) {
-    $log.info('WebkitSpeechRecognitionGuiCtrl');
+    console.log('WebkitSpeechRecognitionGuiCtrl')
 
-    var _self = this;
-    var msg = this.msg;
-    getLanguage = [];
+    var _self = this
+    var msg = this.msg
 
-    continuous = true;
-    clickedFromWebGui = false;
-    this.updateState = function(service) {
-        $scope.service = service;
-
-        getLanguage = service.languagesList;
-
-        $scope.lang = {
-            availableOptions: [],
-            selectedOption: {}
-        };
-        for (var key in getLanguage) {
-            $scope.lang.availableOptions.push({
-                id: key,
-                name: getLanguage[key]
-            });
-        }
-        $scope.lang.selectedOption.id = service.currentWebkitLanguage;
-        $scope.current_language = service.currentWebkitLanguage;
-        continuous = service.continuous;
-
-        if ($scope.service.listening != $scope.recognizing) {
-            $log.info("Change listening state!");
-            // huh!? not initialized  yet?!
-            if (!angular.isUndefined($scope.recognizing)) {
-                // update the recognizing state of the gui.
-                // TODO: fix me. it doesn't handle the stop listening case (likely)
-                if ((!$scope.recognizing && $scope.service.listening) || !clickedFromWebGui) {
-                    // TODO: start Recognition is actually a toggle.. not start.
-                    $scope.startRecognition();
-                }
-            }
-            ;
-        } else {
-            $log.info("State did not change.");
-        }
+    $scope.languages = {
```

Comment:
These likely need to move down to the base java class.. sorry.  if a browser isn't open.. you won't be able to query for what the langauges are and that isn't super.

Reply:
ya .. discussed, we agree, its not quite there - the Java Land Chinese Locales need some work

Reply:
This has been fixed, tested and verified.